sharktank/sharktank/ops/signatures.py

```diff
@@ -824,11 +826,23 @@ def replicate(input: AnyTensor, count: int) -> ShardedTensor:
 
 @replicate.trampoline
 def _replicate_trampoline(
-    d: SignatureDispatcher, input: AnyTensor, count: int
+    d: SignatureDispatcher,
+    input: AnyTensor,
+    count: int,
+    devices: Tuple[int] | None = None,
+    pinned: bool | None = None,
 ) -> ShardedTensor:
     tensors = (input,)
+    if isinstance(input, torch.Tensor):
+        devices = devices if devices is not None else tuple(range(count))
+        pinned = pinned if pinned is not None else False
+    else:
+        # TODO: Is this correct? Will use data on `input`.
+        assert devices is None
+        assert pinned is None
+
     for override in d.find_overrides(tensors):
-        result = override(input, count=count)
+        result = override(input, count=count, devices=devices, pinned=pinned)
```

Comment:
@rsuderman

Reply:
Does not work for tracing sharded llama since incoming tensor is a fake tensor. This behavious may be better handled by doing `ops.replicate(torch.Tensor).clone(pinned=False)` if we wish to override.